src/sentry/tasks/performance_detection.py

```diff
@@ -331,3 +349,51 @@ def visit_span(self, span: Span):
                 fingerprint
             ] >= timedelta(milliseconds=duration_threshold):
                 self.stored_issues[fingerprint] = {"span_id": span_id}
+
+
+class UncompressedAssetSpanDetector(PerformanceDetector):
+    """
+    Checks for large assets that are affecting load time.
+    """
+
+    __slots__ = "stored_issues"
+
+    settings_key = DetectorType.UNCOMPRESSED_ASSET_SPAN
+
+    def init(self):
+        self.stored_issues = {}
+
+    def visit_span(self, span: Span) -> None:
+        op = span.get("op", None)
+        if not op:
+            return
+
+        allowed_span_ops = self.settings.get("allowed_span_ops")
+        if op not in allowed_span_ops:
+            return
+
+        data = span.get("data", None)
+        transfer_size = data and data.get("Transfer Size", None)
+        encoded_body_size = data and data.get("Encoded Body Size", None)
+        decoded_body_size = data and data.get("Decoded Body Size", None)
+        if not (encoded_body_size and decoded_body_size and transfer_size):
+            return
+
+        # Ignore assets from cache, either directly (nothing transferred) or via
+        # a 304 Not Modified response (transfer is smaller than asset size).
+        if transfer_size <= 0 or transfer_size < encoded_body_size:
+            return
+
+        # Ignore assets that are already compressed.
+        if encoded_body_size != decoded_body_size:
```

Comment:
I'm not sure they are always equal anyway, I'd maybe verify iirc `message payload` or `body payload` mean slightly different things, but I can't remember since it's been a while since I've fixed a compression issue. It's either decoded or transfer that can be larger because of headers, I believe.

Reply:
Good callout - this was true in my local testing but I'll double-check the SDK & `Performance` API docs.

Reply:
Confirmed again that these are the same in practice locally, and it seems safe based on my reading of MDN (both [encodedBodySize](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceResourceTiming/encodedBodySize) and [decodedBodySize](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceResourceTiming/decodedBodySize) measure the size of the payload body specifically.)